### PR TITLE
update and delete records

### PR DESCRIPTION
**Pull request by fullstop125**
In the following milestone I managed to implement the following : 

- Add a column `species` of type `string` to your `animals` table. Modify your `schema.sql` file.
- Insert the following data:
  - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- Remember to add these insertions to your `data.sql` file.

----

- Inside a transaction update the `animals` table by setting the `species` column to `unspecified`. Verify that change was made. Then roll back the change and verify that the `species` columns went back to the state before the transaction.
- Inside a transaction:
  - Update the `animals` table by setting the `species` column to `digimon` for all animals that have a name ending in `mon`.
  - Update the `animals` table by setting the `species` column to `pokemon` for all animals that don't have `species` already set.
  - Commit the transaction.
  - Verify that change was made and persists after commit.
- **Take a screenshot of the results of your actions.**
- Now, take a deep breath and... **Inside a transaction** delete all records in the `animals` table, then roll back the transaction.
- After the rollback verify if all records in the `animals` table still exists. After that, you can start breathing as usual ;) 
- **Take a screenshot of the results of your actions.**
- Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
- **Take a screenshot of the results of your actions.**
- Remember to add all these transactions to your `queries.sql` file.

----
- Write queries to answer the following questions:
  - How many animals are there?
  - How many animals have never tried to escape?
  - What is the average weight of animals?
  - Who escapes the most, neutered or not neutered animals?
  - What is the minimum and maximum weight of each type of animal?
  - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
- **Take a screenshot of the results of your queries.**
- 
Screenshot from 2022-10-20 16-43-33
Screenshot from 2022-10-20 16-43-55
Screenshot from 2022-10-20 16-45-04
Screenshot from 2022-10-20 16-45-45
Screenshot from 2022-10-20 16-46-03
Screenshot from 2022-10-20 16-46-23

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -6,5 +6,6 @@ CREATE TABLE animals(
     date_of_birth DATE NOT NULL,
     escape_attempts NUMERIC NOT NULL,
     neutered BOOLEAN NOT NULL,
-    weight_kg DECIMAL NOT NULL
+    weight_kg DECIMAL NOT NULL,
+    species VARCHAR(100)
```
Comment (mikethreels) [inline schema.sql]: - [ ] in order to add a column you are supposed to use the `ALTER TABLE` query

File: data.sql [hunk 2]
```diff
@@ -1,33 +1,52 @@
...
+--         escape_attempts,
+--         neutered,
+--         weight_kg
+--     )
+-- VALUES ('Pikachu', '2021-01-07', '1', 'false', '15.04');
+-- INSERT INTO animals (
+--         name,
+--         date_of_birth,
+--         escape_attempts,
+--         neutered,
```
Comment (mikethreels) [inline data.sql]: - [ ] please do not comment out the previously added data.

File: queries.sql [hunk 3]
```diff
@@ -1,10 +1,58 @@
...
-SELECT * from animals WHERE name != 'Gabumon';
-SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3; 
+-- SELECT name from animals WHERE name LIKE '%mon';
+-- SELECT * from animals WHERE EXTRACT(year from date_of_birth) BETWEEN 2016 AND 2019;
+-- SELECT name from animals WHERE neutered=true AND escape_attempts < 3;
+-- SELECT date_of_birth from animals WHERE name='Agumon' OR name='Pikachu';
+-- SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
+-- SELECT * from animals WHERE neutered = true;
+-- SELECT * from animals WHERE name != 'Gabumon';
+-- SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
```
Comment (mikethreels) [inline queries.sql]: - [ ] no need to comment out the previously created queries.

File: queries.sql [hunk 4]
```diff
@@ -1,10 +1,58 @@
...
+UPDATE animals set species = 'pokemon' WHERE name not like '%mon';
+
+commit;
+
+begin;
+
+DELETE from animals;
+
+rollback;
+
```
Comment (mikethreels) [inline queries.sql]: - [ ] before and after every change you should verify that change were made

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Hassan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @fullstop125        :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for making all the requested changes good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**